source/jormungandr/jormungandr/utils.py

```diff
@@ -494,6 +494,30 @@ def get_pt_object_from_json(dict_pt_object, instance):
     return pt_object
 
 
+def transform_entrypoint(dict_pt_object, uri):
+    if not dict_pt_object:
+        return None
+    if MAP_STRING_PTOBJECT_TYPE.get(dict_pt_object.get("embedded_type")) != type_pb2.ADDRESS:
+        return dict_pt_object
+    first_within_zone = next(
+        (
+            wz
+            for wz in dict_pt_object.get("within_zones", [])
+            if MAP_STRING_PTOBJECT_TYPE.get(wz.get("embedded_type")) == type_pb2.POI
+            and wz.get("poi", {}).get("children", [])
+        ),
+        None,
+    )
+    if not first_within_zone:
+        return dict_pt_object
+    if not is_coord(uri):
+        return dict_pt_object
+    lon, lat = get_lon_lat(uri)
+    first_within_zone["poi"]["coord"]["lon"] = "{}".format(lon)
+    first_within_zone["poi"]["coord"]["lat"] = "{}".format(lat)
```

Comment:
I would put a comment here to express why we're changing the coordinate. Here is a suggestion.

```
# We're putting back the coordinate of the end-user, who is in a POI area (with entrypoints).
# If we don't, barycenter of the POI area will be displayed which means nothing for the end user.
```

Reply:
:+1: Done